packages/react-native-reanimated/src/hook/useAnimatedStyle.ts

```diff
@@ -226,7 +234,15 @@ function styleUpdater(
           animationsActive
         );
         if (finished) {
-          last[propName] = updates[propName];
+          if (Array.isArray(updates[propName])) {
+            updates[propName].forEach((obj: StyleProps) => {
+              for (const prop in obj) {
+                last[propName][prop] = obj[prop];
+              }
+            });
+          } else {
+            last[propName] = updates[propName];
+          }
```

Comment:
Why do we need this?

Reply:
It fixes boxShadow color not properly animating. It was caused by the boxShadow being an object. The `toValue` and `current` were becoming the same value causing the shadow to just switch color

Reply:
So the underlying issue is that `updates[propName]` doesn't have the updated array, yes? If that's true, then why don't we fix it so `updates` would have it right?

Reply:
sorry, i didn't explained it well 😞 

the `toValue` and `current`  end up being the same value because when we previously did `last[propName] = updates[propName]` we again referenced the object, not copied the prop to the `state.last`. So if our `last`['boxShadow']` is holding reference to `updates['boxShadow']`, the `toValue` relies on `last` state, then as soon as current changes, the `toValue` follows resulting in a jump rather than a animation. Not spreading but assigning directly prop to prop gave the expected result

(fields to look at is `current` and `toValue`)
Before:
<img width="1099" alt="image" src="https://github.com/user-attachments/assets/8cdf4724-c3d3-41a0-9aca-edabbf038213" />
https://github.com/user-attachments/assets/9474379d-8a31-4d97-b37d-b59733bf24ee
After:
<img width="1121" alt="image" src="https://github.com/user-attachments/assets/17818ee1-ef8e-47af-a467-1cb37ad829ea" />
https://github.com/user-attachments/assets/3e9e5986-864f-404f-9e95-058a89122dd8

Reply:
I understand now 👍